### PR TITLE
Multitask regression

### PR DESCRIPTION
**Pull request by ElanaPearl**
What's in the PR:
* Adds in code for RobertaForSequenceClassification into repo
* Makes variant of this for RobertaForRegression which handles single or multi-task regression. Also normalizes/unnormalizes the values of each task so we can do multiple tasks on different scales.
* Added regression dataset class that reads in data from a csv, keeps track of all labels, and the mean/std.dev of each task so they can be normalized. This dataset starts by embedding all the smiles which isn't the most efficient.
* Also added a lazy regression dataset which uses hugging face's lazy csv dataset loader but currently has some issues
* Data collator that works for multitask (the vanilla one assumes each label is a single integer)
* Made it so each model just returns logits when called without labels (for eval)

Testing:
* Sequence Classification has been able to easily overfit to a dataset that always has 1 label
* Multitask Regression has been able to overfit to a dataset that has multiple consistent labels

Next steps:
* Switch over to new lazy dataloader
* Get this to work when some labels are NaN
* Reduce the repeated code- maybe merge all training scripts into one since they are quite similar
* Get multi-task regression to learn a simple molecular property (MW maybe?)
* Misc. code cleaning / reorganization (these probably shouldn't all be in a masked-lm folder but don't wanna do too much in 1 PR)

### CODE REVIEW

**Comment by seyonechithrananda** [comment]
Looks good so far! I'll try out the sequence classification script on my own and see how it does

File: chemberta/masked-lm/train_roberta_regression.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+""" Script for training a Roberta Regression Model (single or multi-task)
```
Comment (gabegrand) [inline chemberta/masked-lm/train_roberta_regression.py:1]: There's a lot of code overlap between `train_roberta_regression.py` and `train_roberta_sequenceclass.py` - would you consider refactoring these into a single script?

File: chemberta/utils/raw_text_dataset.py [hunk 2]
```diff
@@ -41,3 +43,66 @@ def __getitem__(self, i):
...
+    def __getitem__(self, idx):
+        item = {key: torch.tensor(val[idx]) for key, val in self.encodings.items()}
+        item['label'] = torch.tensor(self.labels[idx], dtype=torch.float32)
+        return item
+
+    def __len__(self):
+        return len(self.labels)
+
+
+# This dataset is more efficient (supposedly) but has some issues
```
Comment (gabegrand) [inline chemberta/utils/raw_text_dataset.py:74]: What are the issues with the lazy loading?

File: chemberta/masked-lm/train_roberta_regression.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+from transformers import DataCollatorForLanguageModeling
+from transformers import Trainer, TrainingArguments
+
+from tokenizers import ByteLevelBPETokenizer
+
+FLAGS = flags.FLAGS
+
+# RobertaConfig params
+flags.DEFINE_integer(name="vocab_size", default=52_000, help="")
```
Comment (gabegrand) [inline chemberta/masked-lm/train_roberta_regression.py]: Default is very high for SMILES - consider changing

**Review by gabegrand — APPROVED** [review]
Awesome work @ElanaPearl! Great job getting regression/classification to work entirely inside HuggingFace. The overfitting tests you ran give me a high degree of confidence that these models and the normalization logic are both working as expected.

File: chemberta/masked-lm/train_roberta_regression.py (hunk 3 above)
Comment (seyonechithrananda) [reply chemberta/masked-lm/train_roberta_regression.py]: Agreed, I think a 500-1000 vocab_size is better here for BPE models (Smiles-tokenizer vocab is pre-defined with an approx size of 592).

File: chemberta/masked-lm/train_roberta_regression.py (hunk 1 above)
Comment (seyonechithrananda) [reply chemberta/masked-lm/train_roberta_regression.py:1]: I believe we discussed combining both in Slack earlier once normalizing + overfit tests worked, so we should be able to do so! I can take a crack at this if @ElanaPearl is ok with merging this in.

**Comment by seyonechithrananda** [comment]
Could we perhaps split the multi-task reg code into a new subdirectory? In terms of code organization, it would be nice to have a directory for seq-classification/fine-tuning (MoleculeNet), multi-task reg (Pubchem), and masked-lm. 

I'm really excited to try this out, looks awesome. :) Looking forward to seeing how masked-LM and multi-task reg compare.
